commands/peers/addpeer.go

```diff
@@ -43,6 +49,44 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		rest.SendHTTPError(w, http.StatusInternalServerError, *rsp.OpError)
 		return
 	}
+
+	// Add member to etcd server
+	var c etcdclient.Client
+	mAPI := etcdclient.NewMembersAPI(c)
+	//_, e = membersAPI.Add(context.Background(), p.Name)
+	member, e := mAPI.Add(etcdcontext.Background(), p.Name)
+	if e != nil {
+		//rest.SendHTTPError(w, http.StatusInternalServerError, strings.e)
+		rest.SendHTTPError(w, http.StatusInternalServerError, e.Error())
+		return
+	}
+
+	newID := member.ID
+	newName := member.Name
+
+	log.Info("New member named %s with ID %s added to cluster", newName, newID)
+
+	mlist, e := mAPI.List(etcdcontext.Background())
+	if e != nil {
+		rest.SendHTTPError(w, http.StatusInternalServerError, e.Error())
+		return
+	}
+
+	conf := []string{}
+	for _, memb := range mlist {
+		for _, u := range memb.PeerURLs {
+			n := memb.Name
+			if memb.ID == newID {
+				n = newName
+			}
+			conf = append(conf, fmt.Sprintf("%s=%s", n, u))
+		}
+	}
+
+	log.Info("ETCD_NAME=%q", newName)
```

Comment:
Same here. You should be using structured logging. Look at other logs for examples.